nifi-nar-bundles/nifi-box-bundle/nifi-box-processors/src/main/java/org/apache/nifi/processors/box/PutBoxFile.java

```diff
@@ -0,0 +1,366 @@
+        }
+    }
+
+    private BoxFolder.Info getOrCreateFolder(String folderName, String parentFolderId, boolean createFolder) {
+        final Optional<BoxFolder.Info> existingFolder = checkFolderExistence(folderName, parentFolderId);
+
+        if (existingFolder.isPresent()) {
+            return existingFolder.get();
+        }
+
+        if (createFolder) {
+            getLogger().debug("Create folder " + folderName + " parent id: " + parentFolderId);
+
+            final BoxFolder parentFolder = getFolder(parentFolderId);
+            return parentFolder.createFolder(folderName);
+        } else {
+            throw new ProcessException(format("The specified subfolder '%s' does not exist and '%s' is false.", folderName, CREATE_SUBFOLDER.getDisplayName()));
+        }
+    }
+
+    private Optional<BoxFolder.Info> checkFolderExistence(final String folderName, final String parentFolderId) {
+        final BoxFolder parentFolder = getFolder(parentFolderId);
+        return StreamSupport.stream(parentFolder.getChildren("name").spliterator(), false)
+                .filter(BoxFolder.Info.class::isInstance)
+                .map(BoxFolder.Info.class::cast)
+                .filter(info -> info.getName().equals(folderName))
+                .findAny();
+    }
+
+    private Optional<BoxFile.Info> getFileByName(final String filename, final BoxFolder parentFolder) {
+        return StreamSupport.stream(parentFolder.getChildren("name").spliterator(), false)
+                .filter(BoxFile.Info.class::isInstance)
+                .map(BoxFile.Info.class::cast)
+                .filter(info -> info.getName().equals(filename))
+                .findAny();
+    }
+
+    private boolean isMultiLevelFolder(int indexOfPathSeparator, String folderName) {
+        return indexOfPathSeparator > 0 && indexOfPathSeparator < folderName.length() - 1;
+    }
+
+    private void handleUploadError(final String conflictResolution, final String filename, BoxFolder folder, final BoxAPIException e) {
+        if (e.getResponseCode() == CONFLICT_RESPONSE_CODE) {
+            handleConflict(conflictResolution, filename, folder, e);
+        } else {
+            throw new ProcessException(e);
+        }
+    }
+
+    private void handleConflict(final String conflictResolution, final String filename, BoxFolder folder, final BoxAPIException e) {
+        final String path = BoxFileUtils.getPath(folder.getInfo());
+
+        if (IGNORE_RESOLUTION.equals(conflictResolution)) {
+            getLogger().info("File with the same name '{}' already exists in '%s'. Remote file is not modified due to {} being set to '{}'.",
+                    filename, path, CONFLICT_RESOLUTION.getDisplayName(), conflictResolution);
+        } else if (FAIL_RESOLUTION.equals(conflictResolution)) {
+            throw new ProcessException(format("File with the same name '%s' already exists in '%s'.", filename, path), e);
```

Comment:
Recommend avoiding periods at the end of messages:
```suggestion
            throw new ProcessException(format("File with the same name [%s] already exists in [%s]", filename, path), e);
```